assume/units/building.py

```diff
@@ -0,0 +1,404 @@
+                    == self.model.heat_demand[t]
+                    + self.model.dsm_blocks["thermal_storage"].charge_thermal[t]
+                )
+            else:
+                return (
+                    self.model.dsm_blocks["heatpump"].heat_out[t]
+                    + self.model.dsm_blocks["boiler"].heat_out[t]
+                    == self.model.heat_demand[t]
+                )
+
+    def define_sets(self) -> None:
+        """
+        Defines the sets for the Pyomo model.
+        """
+        self.model.time_steps = pyo.Set(
+            initialize=[idx for idx, _ in enumerate(self.index)]
+        )
+
+    def define_parameters(self):
+        """
+        Defines the parameters for the Pyomo model.
+        """
+        self.model.electricity_price = pyo.Param(
+            self.model.time_steps,
+            initialize={t: value for t, value in enumerate(self.electricity_price)},
+        )
+        self.model.natural_gas_price = pyo.Param(
+            self.model.time_steps,
+            initialize={t: value for t, value in enumerate(self.natural_gas_price)},
+        )
+        self.model.heat_demand = pyo.Param(
+            self.model.time_steps,
+            initialize={t: value for t, value in enumerate(self.heat_demand)},
+        )
+        self.model.ev_load_profile = pyo.Param(
+            self.model.time_steps,
+            initialize={t: value for t, value in enumerate(self.ev_load_profile)},
+        )
+        self.model.additional_electricity_load = pyo.Param(
+            self.model.time_steps,
+            initialize={
+                t: value for t, value in enumerate(self.additional_electricity_load)
+            },
+        )
+
+    def define_variables(self):
+        """
+        Defines the variables for the Pyomo model.
+        """
+        self.model.total_power_input = pyo.Var(
+            self.model.time_steps, within=pyo.NonNegativeReals
+        )
+        self.model.variable_cost = pyo.Var(
+            self.model.time_steps, within=pyo.NonNegativeReals
+        )
+
+    def define_constraints(self):
+        @self.model.Constraint(self.model.time_steps)
+        def total_power_input_constraint(m, t):
+            """
+            Ensures the total power input is the sum of power inputs of all components.
+            """
+            return (
+                m.total_power_input[t]
+                == self.model.dsm_blocks["heatpump"].power_in[t]
+                + self.model.dsm_blocks["boiler"].power_in[t]
+                + self.model.dsm_blocks["ev"].charge_ev[t]
+                + self.model.additional_electricity_load[t]
+            )
+
+        @self.model.Constraint(self.model.time_steps)
+        def variable_cost_constraint(m, t):
+            """
+            Calculates the variable cost per time step.
+            """
+            return (
+                self.model.variable_cost[t]
+                == self.model.dsm_blocks["heatpump"].operating_cost_hp[t]
```

Comment:
Here you call `operating_cost_hp` in L:371 just `operating_cost`. Same for other units.
Please check also in dst_components.py. For me just  `operating_cost` is fine but it is on you.

Reply:
@HafnerMichael If you change it to operating_cost then it will have a conflict and delete one variable with the same name. To avoid this I have kelp the variable name unique. Dif you try to run after renaming all the variable names to operating_cost?

Reply:
@Manish-Khanra Yes i have renamed all the variables to just operating_cost. Now I reverted the changes and adapted the inconsistent places and the same error occurs again, if I change the naive strategy to calc the marginal costs of a unit for the price.
I also tried to rename the variables power_in and heat_out of the hp and boiler but still the same...
Does the duplication only apply for variables or also for parameter?

The error:
```
No value for uninitialized NumericValue object dsm_blocks[heatpump].operating_cost_hp[0]
ERROR:pyomo.common.numeric_types:evaluating object as numeric value: dsm_blocks[heatpump].operating_cost_hp[0]
```